src/odemis/acq/stitching/_weaver.py

```diff
@@ -57,12 +56,45 @@ def addTile(self, tile):
         img.mergeMetadata(tile.metadata)
         self.tiles.append(tile)
 
+    @abstractmethod
     def getFullImage(self):
         """
-        return (2D DataArray): same dtype as the tiles, with shape corresponding to the bounding box. 
+        Assembles the tiles into a large image.
+        return (2D DataArray): same dtype as the tiles, with shape corresponding to the bounding box.
         """
+        pass
 
-        tiles = self.tiles
+    def _adjust_brightness(self, tiles):
+        """
+        Adjusts the brightness of a list of tiles, so they all have the same mean value.
+        tiles (2D DataArray): input tiles
+        return (2D DataArray): tiles with adjusted brightness
+        """
+        im_brt = numpy.mean(tiles)
+        for tile in tiles:
+            tile_brt = numpy.mean(tile)
+            diff = numpy.array(im_brt - tile_brt).astype(tile.dtype)  # single value, use array for type cast
+            tile += diff
```

Comment:
There are two issues with this code:
* The original data (`tile`) is changed. In general, you don't want such side effect in any function. It's a little annoying here because that means you'll essentially need 2x more memory. You could avoid this though by adjusting the brightness of the tile a little later, only at the very moment it's used in the copy.
* When summing, you can cause over- and under-flows. You want to do clipping. However I don't know the best way to handle clipping generically in numpy. Probably you could just handle it for integers, and for floats it'll typically work fine. For inspiration, see odemis.util.img.Subtract() and odemis.driver.spectrometer l319. Maybe if you write a nice function, you can put it in odemis.util.img too.